Automattic/wpcom.js
# Upgrade build pipeline and remove n8-make

Removes dependency on `n8-make` and upgrades the build tools to latest versions:
- Babel 7 with CoreJS 2
- Webpack 4
- ESLint 5
- Mocha 6

Babel uses `@wordpress/browserslist-config` to determine target browsers and the set of transforms. Missing platform features are polyfilled with CoreJS2, using the `useBuiltIns: entry` option.

Runtime helpers are transformed to imports from `@babel/runtime`.

There is a minimalistic Webpack configuration to create the bundled `wpcom.js` script that's intended for direct inclusion with a script tag.

Makefile doesn't need `n8-make` and is completely self-contained.

**How to test:**
The output should be almost indentical to the previous version. Best tested by linking the package into Calypso and testing there.

Note that I removed the `add-module-exports` Babel transform. That could break some usages that do the import incorrectly:
```js
import { Me } from 'wpcom';
```
no longer works, because `Me` is a property of the default export, not an exported binding.

We'll bump the major version when publishing, so the breaking change shouldn't be a problem.
